build/Common.prod.props

```diff
@@ -10,6 +10,13 @@
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
   </ItemGroup>
+  <ItemGroup Condition="'$(MinVerTagPrefix)' == 'core-' AND '$(OS)' == 'Windows_NT'">
```

Comment:
Could we run DotnetApiCompat as a CI task instead? This is package is still in preview so we wouldn't want to block our builds because of this. There are already some issues with this package. It incorrectly throws an error when run for the `OpenTelemetry` project. I created an issue to track that: https://github.com/dotnet/arcade/issues/7511

If we could run this as part of a separate CI task, we could mark it optional and wait for the .NET Core Engineering team to fix this. @cijothomas  could mark the CI task as optional until they release a stable version of the package.